### PR TITLE
CBAPI-2552: Fix bugs in helpers.py and increase test coverage

### PR DESCRIPTION
**Pull request by emitreva-cb**
## Pull request checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Tests have been added that prove the fix is effective or that the feature works.
- [x] New and existing tests pass locally with the changes.
- [x] Code follows the style guidelines of this project (PEP8, clean code, linter).
- [x] A self-review of the code has been done.

## What is the ticket or issue number?
https://jira.carbonblack.local/browse/CBAPI-2552

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/cbc_sdk/helpers.py [hunk 1]
```diff
@@ -92,7 +92,7 @@ def get_cb_cloud_object(args):
     return cb
 
 
-def get_object_by_name_or_id(cb, cls, name_field="name", id=None, name=None, force_init=True):
+def get_object_by_name_or_id(cb, cls, name_field="name", id=None, name='', force_init=True):
```
Comment (emitreva-cb) [inline src/cbc_sdk/helpers.py:95]: If name is not provided, then it fails to format it - attempted_to_find = "name {0:s}".format(name) and then fails with the error that  attempted_to_find is used before assignment.

File: src/cbc_sdk/helpers.py [hunk 2]
```diff
@@ -111,7 +111,7 @@ def get_object_by_name_or_id(cb, cls, name_field="name", id=None, name=None, for
     try:
         if id:
             attempted_to_find = "ID of {0:d}".format(id)
-            objs = [cb.select(cls, id, force_init=force_init)]
```
Comment (emitreva-cb) [inline src/cbc_sdk/helpers.py:114]: As far as I saw NewBaseModel has force_init as parameter of the __init__, but not the classes that are inheriting that base class.

**Review by avanbrunt-cb — CHANGES_REQUESTED** [review]
Can we also remove `parser.add_argument("--window", help="Define search window", default='3d')` from `build_cli_parser` as this argument is not needed for every parser. We will need to add it to any example scripts which use it but that should be a small change

File: src/cbc_sdk/helpers.py [hunk 3]
```diff
@@ -111,7 +111,7 @@ def get_object_by_name_or_id(cb, cls, name_field="name", id=None, name=None, for
...
+            objs = [cb.select(cls, id)]
         else:
```
Comment (avanbrunt-cb) [inline src/cbc_sdk/helpers.py:115]: This function requires either an id or name we should through an error neither are provided. I would leave the `name=None` then convert the if else to an if elif else where the else raises an Exception indicating that either id or name should be provided. We should also consider the case where both are provided, do we want to fetch by id then verify if the name matches or raise an error before making an API request indicating only one can be specified

**Review by avanbrunt-cb — APPROVED** [review]
Thanks! Looks good

**Review by hkaragitliev-cb — APPROVED** [review]
LGTM

**Pull request merged**